### DISCUSSION TITLE
TrajOptTaskflow Failure

### DISCUSSION THREAD

**Discussion by mohamedsayed18** [tesseract-robotics/tesseract_planning#100] [Q&A]
```
Error:   TrajOptTaskflow Failure: 
         at line 51 in space/src/motion-planner/tesseract_planning/tesseract_process_managers/src/core/utils.cpp
```
I got this error when I try to do a linear motion from the start position to a point, followed by a linear motion to another point
```
Waypoint wp0 = CartesianWaypoint(getPoses()[0]);
Waypoint wp1 = CartesianWaypoint(getPoses()[1]);
```
These waypoints are VectorIsometry3d poses
```
PlanInstruction plan_f0(wp0, PlanInstructionType::LINEAR, "FREESPACE");
PlanInstruction plan_f1(wp1, PlanInstructionType::LINEAR, "RASTER");
program.push_back(plan_f0);
program.push_back(plan_f1);
```
The program generates a path but it has an unexpected motion between the two points and the path is not straight line
![Two_point_error](https://user-images.githubusercontent.com/27130471/123416011-b6a98e80-d5be-11eb-8bb4-0d42aeace2be.gif)

But when changing the motion of the first instruction from LINEAR to FREESPACE, it works fine and the path is straight
![two_point_clean](https://user-images.githubusercontent.com/27130471/123415986-ae515380-d5be-11eb-9a67-a424cf2f6876.gif)

**Comment by marip8** [comment]
This seems like more of an issue than a general discussion, so I migrated this to #101. Let's continue this conversation on the issue instead
